include/cetl/pmr/interface_ptr.hpp

```diff
@@ -0,0 +1,92 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PMR_INTERFACE_PTR_H_INCLUDED
+#define CETL_PMR_INTERFACE_PTR_H_INCLUDED
+
+#include "cetl/pmr/function.hpp"
+
+#include <memory>
+#include <functional>
+
+namespace cetl
+{
+namespace pmr
+{
+
+template <typename Interface, typename Pmr>
+class PmrInterfaceDeleter final
+{
+public:
+    template <typename PmrAllocator>
+    PmrInterfaceDeleter(PmrAllocator alloc, std::size_t obj_count)
+        : deleter_{alloc.resource(), [alloc, obj_count](Interface* ptr) mutable {
+                       using Concrete     = typename PmrAllocator::value_type;
+                       auto* concrete_ptr = static_cast<Concrete*>(ptr);
+
+                       concrete_ptr->~Concrete();
+                       alloc.deallocate(concrete_ptr, obj_count);
+                   }}
+    {
+    }
+
+    template <typename Down, typename = std::enable_if_t<std::is_base_of<Interface, Down>::value>>
+    PmrInterfaceDeleter(const PmrInterfaceDeleter<Down, Pmr>& other)
+        : deleter_{other.get_memory_resource(), [other](Interface* ptr) {
+                       // Delegate to the down class deleter.
```

Comment:
```suggestion
                       // Delegate to the down class deleter.
                       // The down-casting is assumed to be safe because both the base type
                       // and the derived type are statically known in this context.
                       // This is a possible deviation from AUTOSAR M5-2-3; whether the
                       // type is polymorphic or not is irrelevant in this context.
```